blockchain/state_migration.go

```diff
@@ -97,7 +97,12 @@ func (bc *BlockChain) concurrentRead(db *statedb.Database, quitCh chan struct{},
 func (bc *BlockChain) migrateState(rootHash common.Hash) (returnErr error) {
 	bc.wg.Add(1)
 	defer func() {
```

Comment:
`bc.migrationErr` is not set anywhere.
```suggestion
	bc.migrationErr = nil
	defer func() {
		bc.migrationErr = returnErr
```